### PR TITLE
[RFC] cmd/snap-image: tool for building UC images

### PR DESCRIPTION
**Pull request by bboozzoo**
The series is a follow up on a forum topic https://forum.snapcraft.io/t/snapd-tool-for-building-ubuntu-core-images/12118

The patchset introduces a new command `snap-image` that is a central point for image relation 'actions'. Right now, the command is capable of building a bootable image. This was verified as part of a larger work-in-progress branch for PC and Raspberry Pi UC images.

### CODE REVIEW

File: cmd/snap-image/cmd_volume.go [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+			return nil
+		}
+
+		// system data not defined, add it
+		vol.Structure = append(vol.Structure, gadget.VolumeStructure{
+			Name:       "writable",
+			Size:       estimateSize(size, int64(defaultConstraints.SectorSize)),
+			Role:       "system-data",
+			Label:      "writable",
+			Type:       "83,0FC63DAF-8483-4772-8E79-3D69D8477DE4",
```
Comment (zyga) [inline cmd/snap-image/cmd_volume.go]: Could you add a comment about what the type value here represents?

File: cmd/snap-image/cmd_volume.go [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+	return rootfs, nil
+}
+
+type ImageBuilder struct {
+	workDir string
+	pv      *gadget.PositionedVolume
+}
+
+func NewImageBuilder(workDir string, pv *gadget.PositionedVolume) *ImageBuilder {
+	return &ImageBuilder{
```
Comment (zyga) [inline cmd/snap-image/cmd_volume.go]: Nitpick: this could be a one liner.

File: cmd/snap-image/main.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		builder:  builder,
+		optDescs: optDescs,
+		argDescs: argDescs,
+	}
+	commands = append(commands, info)
+	return info
+}
+
+func parser() *flags.Parser {
+	flagopts := flags.Options(flags.PassDoubleDash | flags.HelpFlag)
```
Comment (zyga) [inline cmd/snap-image/main.go]: Just saying, this is probably duplicated from other commands, perhaps it should live "upstairs".

File: cmd/snap-image/main.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+			arg.Description = desc
+		}
+	}
+
+	return parser
+}
+
+func main() {
+	if err := logger.SimpleSetup(); err != nil {
+		fmt.Fprintf(Stderr, "failed to prepare logging: %v\n", err)
```
Comment (zyga) [inline cmd/snap-image/main.go]:
```suggestion
		fmt.Fprintf(Stderr, "cannot to prepare logging: %v\n", err)
```

**Comment by anonymouse64** [comment]
@bboozzoo what's the status of this? do you want me to review this?

**Comment by bboozzoo** [comment]
@anonymouse64 yes, please do. We haven't agreed on the name of the tool yet, but @cmatsuoka has already pulled in most of the patches into #7470

**Comment by bboozzoo** [comment]
Closing until we decide what to do with the tool and how it fits in the larger picture

**Pull request closed without merging**